celery/backends/mongodb.py

```diff
@@ -43,6 +46,33 @@ def __init__(self, **kw):
         self.__dict__.update(kw)
 
 
+def auto_retry(f):
+    ''' Forces decorated function to retry if mongo throws error '''
```

Comment:
This should probably be more generic.
I don't think that merging something that is specific to mongo exceptions is useful.